text_formatter.go

```diff
@@ -74,8 +77,19 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 
 	prefixFieldClashes(entry.Data)
 
-	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
-	isColored := (f.ForceColors || isColorTerminal) && !f.DisableColors
+	f.terminalLock.RLock()
+	terminalDetermined := f.terminalDetermined
+	isTerminal := f.isTerminal
+	f.terminalLock.RUnlock()
+
+	if !terminalDetermined {
+		f.terminalLock.Lock()
+		f.isTerminal = IsTerminal(entry.Logger.Out)
+		f.terminalDetermined = true
+		f.terminalLock.Unlock()
+	}
+
```

Comment:
`isTerminal` needs to be set here since `terminalDetermined` will be false the first time